### ISSUE TITLE
undefined symbol: _ZN12libfreenect25FrameC1EmmmPh

### ISSUE THREAD

**Issue by thomiel** [madadam/freenect2.net#2]
When building the solution I fot a `freenect2.net/Example/bin/Debug/libfreenect2c.so: undefined symbol: _ZN12libfreenect25FrameC1EmmmPh`

Any ideas?

**Comment by thomiel** [comment]
I found an old version of the libfreenect2.so in the base folder that caused the undefined symbol (expected ~/freenect2/lib to be used instead). I just copied the newer libfreenect2.so along with libfreenect2.so.0.2 and libfreenect2.so.0.2.0 into the base folder for now.

Here is a list of tasks I needed to do before I was able to build the solution:
1. Changed the target framework of the Freenect2 subproject from 3.5 to 4.5.
2. I also needed to expose the include path to _make_ the libfreenect2c.so in the native folder:
   export CPLUS_INCLUDE_PATH=~/freenect2/include
   (or CPLUS_INCLUDE_PATH=$CPLUS_INCLUDE_PATH:~/freenect2/include if you are already using this environment variable)
3. Add the 
   export LD_LIBRARY_PATH=$LD_LIBRARY_PATH:~/freenect2/lib
4. finally, after adding the library path in 3., start the solution from the same bash, so C# can find the library via $LD_LIBRARY_PATH (here using MonoDevelop as development environment):
   monodevelop Freenect2.sln
5. Press F5

Good job, madadam, building that wrapper. Maybe you want to add these as build steps to your README.md...

**Comment by madadam** [comment]
Hi there,

Sorry for not responding earlier, but I'm currently busy with other projects. I'll get back to this as soon as I can!

**Comment by RajaMOeez** [comment]
Hello there @madadam  
SORRY for that Question but Im newby with kinect and C# how to build this program ? To use in C#

I have tried to build your code it is not working. 
**here is the error thats what I got after building**   
### 
Error	1	Metadata file 'C:\Users\MOeez Raja\Desktop\freenect2.net-master\freenect2.net-master\bin\Debug\Freenect2DotNet.dll' could not be found	C:\Users\MOeez Raja\Desktop\freenect2.net-master\freenect2.net-master\Example\CSC	Example

any help ,  I'll be very grateful to you.

**Comment by thomiel** [comment]
Hi Raja!

Maybe you will find the instructions on my fork more comprehensive:

https://github.com/thomiel/freenect2.net

**Comment by RajaMOeez** [comment]
@thomiel  so nice of you.

**Comment by madadam** [comment]
Hey @RajaMOeez , if you are on windows, you might potentially be better off using the official Microsoft SDK for Kinect. I wrote this library for linux and never bothered porting it to windows. Or check @thomiel 's fork out, he might have added windows support.

@thomiel : sorry man it took me so long to get back to you. I can't maintain this library anymore. If you are interested, would you mind taking over the reins? I can put a notice in the README that you are now the official maintainer. What do you say?

**Comment by RajaMOeez** [comment]
@madadam  thanks for the quick response.

**Comment by thomiel** [comment]
I didn't care about Windows support. On Windows, I also would use the Microsoft SDK.

A link in the readme would be nice. I still plan to do stuff for it, e.g. calibration.

**Comment by RajaMOeez** [comment]
sorry to ask that nonsense question but Im a newby student of the first semester and I have assignment to solve  a problem with kinect v2. I need to use the C# wrapper to solve a problem basically I need to detect the color from kinect v2 after applying color filter on kinect V2 image frame using Aforge.net library. I have tried it remains failed and got this 
**ERROR** 
**An unhandled exception of type 'System.DllNotFoundException' occurred in AForge.Video.Kinect.dll

Additional information: Unable to load DLL 'freenect': The specified module could not be found. (Exception from HRESULT: 0x8007007E)**

If any help and guidance I'll be very thankful

**Comment by thomiel** [comment]
Have you installed the freenect2 library? Freenect2.NET is just a wrapper for mono.

Which fork do you actually use? Mine or madadams?

**Comment by RajaMOeez** [comment]
yes sir I have already install the Libfreenect2 library. The fork I have been using is https://github.com/OpenKinect/libfreenect2
thanks for the response @thomiel 
Best Regards 
From MOeez

**Comment by thomiel** [comment]
I meant the fork of the .NET wrapper you use.

(don't call me sir, I'm no knight or something)

**Comment by RajaMOeez** [comment]
@thomiel  **Sorry** about that. The fork that I was use is (https://github.com/thomiel/freenect2.net).

What thing do you have use for make this fork. I was trying to do **MAKE** it with **Windows Shell**. but I can't **MAKE** it because the commands that you have gave in this fork, to **MAKE** this fork is not OK with **Windows Shell**

**Comment by thomiel** [comment]
Sorry, Raja. As I said: "I didn't care about Windows support. On Windows, I also would use the Microsoft SDK."

As a first semester newby, your skills might not be sufficient to adapt the .NET wrapper for use with Windows OS in reasonable time. The best way -- supposed you appreciate the open source way -- would be to get the MinGW environment. A GNU version of the _make_ utility is included there. You would have to adapt to all the OS differences, e.g. using .dll files rather than .so files.

Not as steep to climb for a noob: the Windows SDK https://www.microsoft.com/en-us/download/details.aspx?id=44561

**Comment by RajaMOeez** [comment]
thanks for the your response and guidance. I'll try.
